database_Storage/controllers/experienceController.js

```diff
@@ -0,0 +1,60 @@
+
+const asyncHandler = require('express-async-handler');
+const Experience = require('../model/experienceModel');
+
+const getExperience = asyncHandler(async (req, res) => {
+  const experiences = await Experience.find();
+  res.json(experiences);
+});
+
+const setExperience = asyncHandler(async (req, res) => {
+  const { company, startDate, endDate, description } = req.body;
+
+  const experience = await Experience.create({
+    company,
+    startDate,
+    endDate,
+    description,
+  });
+
+  res.status(201).json({ message: 'Experience created successfully', data: experience });
+});
+
+const updateExperience = asyncHandler(async (req, res) => {
+  const { company, startDate, endDate, description } = req.body;
+
+  const experience = await Experience.findById(req.params.id);
+
+  if (experience) {
+    experience.company = company || experience.company;
+    experience.startDate = startDate || experience.startDate;
+    experience.endDate = endDate || experience.endDate;
+    experience.description = description || experience.description;
+
```

Comment:
this is a bug here. for example if u added an experience item with description "ABC", then removed the description i.e its an empty string now, the update request will succede but the description will default back to ABC